File: app/src/main/java/com/sesolutions/responses/music/Images.java
package com.sesolutions.responses.music;

import com.google.gson.annotations.SerializedName;

public class Images {
    @SerializedName("main")
    private String main;
    @SerializedName("icon")
    private String icon;
    @SerializedName("normal")
    private String normal;
    @SerializedName("profile")
    private String profile;

    public String getMain() {
        return main;
    }

    public void setMain(String main) {
        this.main = main;
    }

    public String getIcon() {
        return icon;
    }

    public void setIcon(String icon) {
        this.icon = icon;
    }

    public String getNormal() {
        if (null != normal) {
            return normal;
        } else return getMain();
    }

    public void setNormal(String normal) {
        this.normal = normal;
    }

    public String getProfile() {

        if (null != profile) {
            return profile;
        } else return getNormal();
    }

    public void setProfile(String profile) {
        this.profile = profile;
    }
}

File: app/src/main/java/com/sesolutions/responses/feed/Images.kt
package com.sesolutions.responses.feed

import com.google.gson.annotations.SerializedName

class Images {
    val images : Images? = null
    @SerializedName("main")
    private var main: String? = null
    @SerializedName("icon")
    var icon: String? = null
        get() = if (null != field)
            field
        else
            getNormal()
    @SerializedName("normal")
    private var normal: String? = null
    @SerializedName("title")
    private var title: String? = null
    private val profile: String? = null

    @SerializedName("normalmain")
    private val normalmain: String? = null

    @SerializedName("url")
    private val url: String? = null

    val attachmentData: AttachmentData? = null

    fun getProfile(): String? {
        return profile ?: if (null != normal) {
            normal
        } else
            main

    }

    fun getUrl(): String? {
        return if (null != url) url else main
    }

    fun getNormal(): String? {
        return if (null != normal) normal else main
    }

    fun getMain(): String? {
        return if (null == main) normalmain else main
    }
    fun gettitle(): String?{
        return title
    }

    fun setMain(main: String) {
        this.main = main
        this.normal = main
    }

    fun setMainImage(main: String): Images {
        this.main = main
        this.normal = main
        return this
    }


    fun setNormal(normal: String) {
        this.normal = normal
    }
}
